testbed/HIL_Firmware/VCU_HIL/VCU_HIL/userInit.c

```diff
@@ -0,0 +1,184 @@
+    } 
+    else 
+    {
+        printf("Failed to start TWAI driver\n");
+        return ESP_FAIL;
+    }
+}
+
+
+int spi_init(void)
+{
+    printf("Initializing SPI bus\n");
+
+    esp_err_t ret = ESP_OK;
+
+    memset(&throttle_A, 0, sizeof(spi_device_handle_t));
+    memset(&throttle_B, 0, sizeof(spi_device_handle_t));
+    memset(&brake_pos, 0, sizeof(spi_device_handle_t));
+    memset(&steer_raw, 0, sizeof(spi_device_handle_t));
+
+    spi_bus_config_t buscfg={
+        .mosi_io_num=SPI_MOSI_PIN,
+        .sclk_io_num=SPI_CLK_PIN,
+        .quadwp_io_num=-1,
+        .quadhd_io_num=-1,
+    };
+
+    spi_device_interface_config_t throttleACfg={
+        .clock_speed_hz=20*HZ_PER_MHZ,           //Clock out at 20 MHz
+        .mode=0,                                //SPI mode 0
+        .spics_io_num=THROTTLE_A_CS,            //CS pin
+        .queue_size=7,                          //We want to be able to queue 7 transactions at a time
+    };
+
+    spi_device_interface_config_t throttleBCfg={
+        .clock_speed_hz=20*HZ_PER_MHZ,           //Clock out at 20 MHz
+        .mode=0,                                //SPI mode 0
+        .spics_io_num=THROTTLE_B_CS,            //CS pin
+        .queue_size=7,                          //We want to be able to queue 7 transactions at a time
+    };
+
+    spi_device_interface_config_t brakePosCfg={
+        .clock_speed_hz=20*HZ_PER_MHZ,           //Clock out at 20 MHz
+        .mode=0,                                //SPI mode 0
+        .spics_io_num=BRAKE_POS_CS,               //CS pin
+        .queue_size=7,                          //We want to be able to queue 7 transactions at a time
+    };
+
+    spi_device_interface_config_t steerRawCfg={
+        .clock_speed_hz=20*HZ_PER_MHZ,           //Clock out at 10 MHz
+        .mode=0,                                //SPI mode 0
+        .spics_io_num=STEER_RAW_CS,               //CS pin
+        .queue_size=7,                          //We want to be able to queue 7 transactions at a time
+    };
+
+    ret=spi_bus_initialize(SPI2_HOST, &buscfg, SPI_DMA_CH_AUTO);
+    ESP_ERROR_CHECK(ret);
+    if(ret != ESP_OK){
+        printf("failed to init SPI bus\r\n");
+    }
+
+    ret=spi_bus_add_device(SPI2_HOST, &throttleACfg, &throttle_A);
+    ESP_ERROR_CHECK(ret);
+    if(ret != ESP_OK){
+        printf("failed to init throttle A DAC\r\n");
+    }
+
+    ret=spi_bus_add_device(SPI2_HOST, &throttleBCfg, &throttle_B);
+    ESP_ERROR_CHECK(ret);
+    if(ret != ESP_OK){
+        printf("failed to init throttle B DAC\r\n");
+    }
+
+    ret=spi_bus_add_device(SPI2_HOST, &brakePosCfg, &brake_pos);
+    ESP_ERROR_CHECK(ret);
+    if(ret != ESP_OK){
+        printf("failed to init brake position DAC\r\n");
+    }
+
+    ret=spi_bus_add_device(SPI2_HOST, &steerRawCfg, &steer_raw);
+    ESP_ERROR_CHECK(ret);
+    if(ret != ESP_OK){
+        printf("failed to init steering raw DAC\r\n");
+    }
+
+    set6551Voltage(0,DacId_throttleA);
```

Comment:
You should probably check return value of this, print error and return error if it fails